src/carnot/planner/compiler/analyzer/convert_metadata_rule.cc

```diff
@@ -85,17 +91,38 @@ StatusOr<bool> ConvertMetadataRule::Apply(IRNode* ir_node) {
   PX_ASSIGN_OR_RETURN(auto parent, metadata->ReferencedOperator());
   PX_ASSIGN_OR_RETURN(auto containing_ops, metadata->ContainingOperators());
 
+  auto resolved_table_type = parent->resolved_table_type();
   PX_ASSIGN_OR_RETURN(std::string key_column_name,
-                      FindKeyColumn(parent->resolved_table_type(), md_property, ir_node));
+                      FindKeyColumn(resolved_table_type, md_property, ir_node));
 
   PX_ASSIGN_OR_RETURN(ColumnIR * key_column,
                       graph->CreateNode<ColumnIR>(ir_node->ast(), key_column_name, parent_op_idx));
 
   PX_ASSIGN_OR_RETURN(std::string func_name, md_property->UDFName(key_column_name));
-  PX_ASSIGN_OR_RETURN(
-      FuncIR * conversion_func,
-      graph->CreateNode<FuncIR>(ir_node->ast(), FuncIR::Op{FuncIR::Opcode::non_op, "", func_name},
-                                std::vector<ExpressionIR*>{key_column}));
+  auto backup_conversion_available = CheckBackupConversionAvailable(resolved_table_type, func_name);
+
+  FuncIR* conversion_func;
+
+  // TODO(ddelnano): Until the short lived process issue (gh#1638) is resolved, use a
+  // conversion function that uses local_addr for pod lookups when the upid based default
+  // (upid_to_pod_name) fails.
+  if (backup_conversion_available) {
+    func_name = "_upid_to_podname_local_addr_fallback";
```

Comment:
Nit: Feels a little weird to put something so specific to the local_addr stuff in here.
To make it a little cleaner, can we pull this into a `AddBackupConversions` function and put this logic into there instead?

Reply:
I took a stab at this in 6f8ee0e41afc2e32eff246a6f1341022e9a72100.

If we were to continue to expand on this pattern, I think we should update the MetadataProperty class to work with a composite lookup key (`time_`, `local_addr` and `upid` => pod name as well as `upid` => pod name). That would allow this rule's logic to stay the same and keep the metadata lookup mappings all in the [metadata_handler.cc file](https://github.com/pixie-io/pixie/blob/1f6d18bfba7bd3cedf99e3d7eec5245b0e018f21/src/carnot/planner/metadata/metadata_handler.cc#L57-L58).

I did investigate what I described above, but I opted for this approach since that "composite lookup" hasn't seemed like a more common pattern at this time.